src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -891,7 +891,9 @@ private TypeGenerationSpec ParseTypeGenerationSpec(in TypeToGenerate typeToGener
                             // property is static or an indexer
                             propertyInfo.IsStatic || propertyInfo.Parameters.Length > 0 ||
                             // It is overridden by a derived property
-                            PropertyIsOverriddenAndIgnored(propertyInfo, state.IgnoredMembers))
+                            PropertyIsOverriddenAndIgnored(propertyInfo, state.IgnoredMembers) ||
```

Comment:
There's an equivalent implementation in the reflection resolver:

https://github.com/dotnet/runtime/blob/3a382f2c51f02bc88869f4d05bfdb9d3c1f8e166/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.Helpers.cs#L141

Should this be updated as well?

Reply:
Yes, we should keep both implementations aligned. For reflection flow, the deduplication is done at one place for both get and set operations, whereas for the source generation flow the fast-path and initializers deduplication logic were split. With this PR it is handled at the same place for all flows.

Reply:
Taking a closer look at the implementation -- this method ultimately calls into the `AddPropertyWithConflictResolution` method which handles conflict resolution (including shadowed properties):

https://github.com/dotnet/runtime/blob/b94a82d63580775b2adb0ccac478719b8cd7049b/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L1037-L1039

My impression is that the bug is caused by a missed case in that logic, so it seems to me that it's this particular logic that should be debugged. 

If you don't know how to debug source generator code, the easiest way to do that is by adding a unit test in `System.Text.Json.SourceGeneration.Roslyn*.Unit.Tests` and debugging them.